bindings/rust/s2n-tls/src/pool.rs

```diff
@@ -287,4 +302,20 @@ mod tests {
         let _: PooledConnection = PooledConnection::new(&pool)?;
         Ok(())
     }
+
+    #[test]
+    fn dereferencing_pooled_connection() -> Result<(), Box<dyn std::error::Error>> {
+        let config_pool = ConfigPoolBuilder::new(Mode::Server, Config::default()).build();
+
+        let pooled_conn: PooledConnection<ConfigPoolRef> = PooledConnection::new(&config_pool)?;
+        let conn = pooled_conn.deref();
+        assert_eq!(pooled_conn.conn.as_ref().unwrap(), conn);
+
+        // We can't test deref_mut() the way that we test deref() above since the borrow checker will only
+        // allow one mutable reference to exist in a scope.
+        // However, the implementation of deref_mut() and deref() are identical aside from the mutability,
+        // so just testing deref() is good enough.
```

Comment:
rather than asserting equivalence of the connection it would be better to just try to use a method through the automatic dereference and assert that the returned values are the same

```suggestion
        let waker = futures_test::task::new_count_waker();
        pooled_conn.set_waker(Some(waker.clone()));
        assert!(pooled_conn.waker().unwrap().will_wake(&waker));
```

Reply:
This sounds fine to me, I'll post a revision with this change.